source/code/source/Entity.h.ejs

```diff
@@ -82,3 +110,83 @@ struct PlayFabEntity
 
     PlayFab::SharedPtr<PlayFab::Entity> entity;
 };
+
+namespace PlayFab
+{
+
+//------------------------------------------------------------------------------
+// Template implementations
+//------------------------------------------------------------------------------
+
+template<typename CallT>
+auto Entity::MakePlayFabCallWithAuthRetry(const TaskQueue& queue, CallT call) -> typename std::result_of_t<CallT(const TaskQueue&)>
+{
+    using AsyncOpT = typename std::result_of_t<CallT(const TaskQueue&)>;
+    using ResultT = typename Detail::UnwrapAsyncT<typename std::result_of_t<CallT(const TaskQueue&)>>;
+
+    class Operation
+    {
+    public:
+        static AsyncOpT Begin(SharedPtr<Entity> entity, const TaskQueue& queue, CallT&& call)
+        {
+            UniquePtr<Operation> operation{ new (Allocator<Operation>{}.allocate(1)) Operation(std::move(entity), queue, std::move(call)) };
+
+            operation->m_call(operation->m_queue).Finally(std::bind(&Operation::HandleCallResult, operation.get(), std::placeholders::_1));
+
+            // Release operation and reclaim upon completion
+            operation.release();
+
+            return operation->m_asyncContext;
+        }
+
+    private:
+        Operation(SharedPtr<Entity> entity, const TaskQueue& queue, CallT&& call) :
+            m_entity{ std::move(entity) },
+            m_queue{ queue.DeriveWorkerQueue() },
+            m_call{ std::move(call) },
+            m_asyncContext{ MakeShared<AsyncOpContext<ResultT>>() }
+        {
+        }
+
+        void HandleCallResult(Result<ResultT>&& result)
+        {
+            // TODO figure out what is actually returned when an EntityToken or SessionTicket has expired
+            // TODO ensure we don't loop multiple times here (i.e. only reauthenticate once)

```

Comment:
Still figuring out how to test this path. Not sure exactly what is returned when a token expires.